runtime/parachains/src/paras_inherent.rs

```diff
@@ -141,7 +143,7 @@ fn backed_candidate_weight<T: frame_system::Config + Config>(
 	<<T as Config>::WeightInfo as WeightInfo>::enter_backed_candidates_variable(
 		candidate.validity_votes.len() as u32,
 	)
-	.saturating_add(if candidate.candidate.commitments.new_validation_code.is_some() {
+	.max(if candidate.candidate.commitments.new_validation_code.is_some() {
```

Comment:
I don't quite get the `max` here? It would at least mandate a comment.

Reply:
I can probably write the code in a more idiomatic way - but basically if there is `new_validation_code` we have a separate weight that we can use. So to get the worst case weight for the candidate we just max two. 

In other words adding them does not make sense because that would give the weight of a `weight_backed_candidate + weigh_backed_candidate_with_validation_code` but we actually just want the worst case weight for the one candidate

Reply:
I changed the logic to a simple if else 9dc1abbf096889c2bbe1f4cdabb818551a9a2834

Reply:
I see, makes sense :+1: